src/System.Linq/tests/UnionTests.cs

```diff
@@ -189,6 +238,44 @@ public void EachHasRepeatsBetweenAndAmongstThemselves()
         }
 
         [Fact]
+        public void EachHasRepeatsBetweenAndAmongstThemselvesMultipleUnions()
+        {
+            int?[] first = { 1, 2, 3, 4, null, 5, 1 };
+            int?[] second = { 6, 2, 3, 4, 5, 6 };
+            int?[] third = { 2, 8, 2, 3, 2, 8 };
+            int?[] fourth = { null, 1, 7, 2, 7 };
+            int?[] expected = { 1, 2, 3, 4, null, 5, 6, 8, 7 };
+
+            Assert.Equal(expected, first.Union(second).Union(third).Union(fourth));
+        }
+
+        [Fact]
+        public void EachHasRepeatsBetweenAndAmongstThemselvesMultipleUnionsCustomComparer()
+        {
+            int?[] first = { 1, 2, 3, 4, null, 5, 1 };
+            int?[] second = { 6, 2, 3, 4, 5, 6 };
+            int?[] third = { 2, 8, 2, 3, 2, 8 };
+            int?[] fourth = { null, 1, 7, 2, 7 };
+            int?[] expected = { 1, 2, 3, 4, null, 5, 6, 8, 7 };
+
+            Assert.Equal(expected, first.Union(second, new Modulo100EqualityComparer()).Union(third, new Modulo100EqualityComparer()).Union(fourth, new Modulo100EqualityComparer()));
```

Comment:
All the input values are less than 100... what are we really validating with these equality comparers?